Project/Wikode/wikodeApp/views.py

```diff
@@ -281,7 +282,101 @@ def getArticles(request):
+    followee_list = getFolloweeList(user)
+
+    context = {
+        'user': user,
+        'follower_list': follower_list,
+        'followee_list': followee_list,
+    }
+
+    return render(request, 'wikodeApp/profilePage.html', context)
+
+
+## Renders the profilePage.html with the clicked user's id information as pk
+## Navigates to /profile/# url.
+@login_required
+def getProfilePageOfOtherfUser(request, pk):
+    ## TODO
+    ## pk arguement may be a unique random 6 digit number that represents the requested user.
+    ## Here we need to convert the unique random number to user id. Or have another number that represents user.
+    ## For development purpose, pk is hardcoded below.
+    other_user = User.objects.get(id=pk)
+    session_user = User.objects.get(id=request.user.id)
+
+    if other_user == session_user:
+        return redirect('wikodeApp:myProfilePage')
+
+    is_followed = FollowRelation.objects.filter(followee_id=other_user.id, follower_id=session_user.id).exists()
+
+    follower_list = getFollowerList(other_user)
+    followee_list = getFolloweeList(other_user)
+
+    context = {
+        'profile': other_user,
+        'is_followed': is_followed,
+        'follower_list': follower_list,
+        'followee_list': followee_list,
+    }
+
+    return render(request, 'wikodeApp/profilePage.html', context)
+
+
+@login_required
+def followUser(request, pk):
+    ## TODO
+    ## pk arguement may be a unique random 6 digit number that represents the requested user.
+    ## Here we need to convert the unique random number to user id. Or have another number that represents user.
+    ## For development purpose, pk is hardcoded below.
+    other_user = User.objects.get(id=pk)
+    session_user = User.objects.get(id=request.user.id)
+
+    is_followed = FollowRelation.objects.filter(followee_id=other_user.id, follower_id=session_user.id).exists()
+
+    if is_followed:
+        following = FollowRelation.objects.get(follower_id=session_user.id, followee_id=other_user.id)
+        following.delete()
+    else:
+        FollowRelation.objects.create(follower_id=session_user.id, followee_id=other_user.id)
+
+
+    ## Return Follow/Unfollow button appearance is determined by is_followed value.
+    ## If True don't show Follow button, show Unfollow instead.
+    return redirect('wikodeApp:getProfilePageOfOtherfUser', pk)
+
+
+## Create a followee list from Follow relation that the follower id is other_user.id
+## List example: [1, 'Can', 'Dayan']
+def getFolloweeList(user):
+    followee_list = json.dumps(
+        list(
+            FollowRelation
+                .objects
+                .filter(follower_id=user.id)
+                .values_list('followee_id', 'followee__first_name', 'followee__last_name')
+        )
+    )
+    return followee_list
+
+
+## Create a follower list from Follow relation that the followee id is other_user.id
+## List example: [1, 'Can', 'Dayan']
+def getFollowerList(user):
```

Comment:
move this to utils under a appropriate user manager file

Reply:
followManager created inside utils. Functions are moved there.